internal/host/plugin/host_catalog.go

```diff
@@ -58,16 +57,16 @@ func allocHostCatalog() *HostCatalog {
 // secret.  The secret shallow copied.
 func (c *HostCatalog) clone() *HostCatalog {
 	cp := proto.Clone(c.HostCatalog)
+	newSecret := proto.Clone(c.secrets)
 
 	hc := &HostCatalog{
 		HostCatalog: cp.(*store.HostCatalog),
+		secrets:     newSecret.(*structpb.Struct),
 	}
-	if c.secrets != nil {
-		newSecret := make(map[string]interface{}, len(c.secrets))
-		for k, v := range c.secrets {
-			newSecret[k] = v
-		}
-		hc.secrets = newSecret
+	// proto.Clone will convert slices with length and capacity of 0 to nil.
+	// Fix this since gorm treats empty slices differently than nil.
+	if c.Attributes != nil && len(c.Attributes) == 0 && hc.Attributes == nil {
```

Comment:
This check is different between here and host set below (in host sets there is no zero-length check).

Reply:
Woops, thank you for catching that.